provider/pkg/functions/insights_account_test.go

```diff
@@ -0,0 +1,292 @@
+// Copyright 2016-2025, Pulumi Corporation.
+
+package functions
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/pulumi/pulumi-go-provider/infer"
+	"github.com/pulumi/pulumi-pulumiservice/provider/pkg/config"
+	"github.com/pulumi/pulumi-pulumiservice/provider/pkg/pulumiapi"
+	"github.com/pulumi/pulumi-pulumiservice/provider/pkg/resources"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type insightsAccountClientMock struct {
+	config.Client
+	getInsightsAccountFunc   func(ctx context.Context, orgName string, accountName string) (*pulumiapi.InsightsAccount, error)
+	listInsightsAccountsFunc func(ctx context.Context, orgName string) ([]pulumiapi.InsightsAccount, error)
+}
+
+func (c *insightsAccountClientMock) GetInsightsAccount(ctx context.Context, orgName string, accountName string) (*pulumiapi.InsightsAccount, error) {
+	if c.getInsightsAccountFunc == nil {
+		return nil, nil
+	}
+	return c.getInsightsAccountFunc(ctx, orgName, accountName)
+}
+
+func (c *insightsAccountClientMock) ListInsightsAccounts(ctx context.Context, orgName string) ([]pulumiapi.InsightsAccount, error) {
+	if c.listInsightsAccountsFunc == nil {
+		return nil, nil
+	}
+	return c.listInsightsAccountsFunc(ctx, orgName)
+}
+
+func TestGetInsightsAccountFunction(t *testing.T) {
+	t.Parallel()
+
+	t.Run("successfully gets an account", func(t *testing.T) {
+		t.Parallel()
+		testOrg := "test-org"
+		testAccount := "test-account"
+		insightsAccount := &pulumiapi.InsightsAccount{
+			ID:                   "account-id-123",
+			Name:                 testAccount,
+			Provider:             string(resources.CloudProviderAWS),
+			ProviderEnvRef:       "test-env",
+			ScheduledScanEnabled: true,
+			ProviderConfig: map[string]interface{}{
+				"regions": []interface{}{"us-west-2"},
+			},
+		}
+		mockedClient := &insightsAccountClientMock{
+			getInsightsAccountFunc: func(ctx context.Context, orgName string, accountName string) (*pulumiapi.InsightsAccount, error) {
+				assert.Equal(t, testOrg, orgName)
+				assert.Equal(t, testAccount, accountName)
+				return insightsAccount, nil
+			},
+		}
+
+		ctx := config.WithMockClient(context.Background(), mockedClient)
```

Comment:
Never use `context.Background()` in a test unless you have a comment explaining why you want the context to live beyond the scope of the test.
```suggestion
		ctx := config.WithMockClient(t.Context(), mockedClient)
```

This feedback applies to the whole file.